### ISSUE TITLE
No TLS for MQTT

### ISSUE THREAD

**Issue by M0uuf** [merlinschumacher/Basecamp#8]
async-mqtt-client seems to be reffering to AsyncTCP for ESP32 for TLS support. At the moment, this is not part of the ESP32 implementation (the ESP8266 has TLS support).
If there is a way to activate TLS support pleas let me know. So long I sadly can not use Basecamp, which is sad since it seams to be a promising platform for an easy project start. However, my sensors have to upload the data to a server on the internet and I will no transmit credentials without TLS.

Kind regards,
André

**Comment by merlinschumacher** [comment]
I haven't yet tested the TLS-Support of async-mqtt-client, but as far as i know it should work on the ESP32. There was simply no time yet to integrate proper support for TLS. But it is on my list.

**Comment by M0uuf** [comment]
Thanks for looking into it.

**Comment by h3ndrik** [comment]
afaik not possible (yet). see:

https://github.com/marvinroger/async-mqtt-client/issues/62#issuecomment-352162662

**Comment by M0uuf** [comment]
Yup, "fatal error: tcp_axtls.h: No such file or directory" is the same error I get, wenn I try to activate TLS.

**Comment by merlinschumacher** [comment]
As soon as AsyncTCP has support for SSL I will patch it into Basecamp.

**Comment by AKozhuhov** [comment]
Maybe it's possible to use this fork? https://github.com/fremouw/AsyncTCP/tree/mbed-tls

**Comment by dpunzeng** [comment]
In the meantime I have abandoned this library matching battle and moved on to microyhython - available on many hw platforms
